core/frontend/src/render/primitives/VertexTableSplitter.ts

```diff
@@ -0,0 +1,614 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Rendering
+ */
+
+import { assert, Constructor, Id64 } from "@itwin/core-bentley";
+import { ColorDef, PackedFeatureTable } from "@itwin/core-common";
+import {
+  computeDimensions, MeshParams, VertexIndices, VertexTable, VertexTableProps, VertexTableWithIndices,
+} from "./VertexTable";
+import { PointStringParams } from "./PointStringParams";
+import { PolylineParams, TesselatedPolyline } from "./PolylineParams";
+import { EdgeParams } from "./EdgeParams";
+
+export type ComputeNodeId = (elementId: Id64.Uint32Pair) => number;
+
+interface TypedArrayBuilderOptions {
+  growthFactor?: number;
+  initialCapacity?: number;
+}
+
+class TypedArrayBuilder<T extends Uint8Array | Uint16Array | Uint32Array> {
+  protected readonly _constructor: Constructor<T>;
+  protected _data: T;
+  protected _length: number;
+  protected readonly _growthFactor: number;
+
+  protected constructor(constructor: Constructor<T>, options?: TypedArrayBuilderOptions) {
+    this._constructor = constructor;
+    this._data = new constructor(options?.initialCapacity ?? 0);
+    this._growthFactor = options?.growthFactor ?? 1.5;
+    this._length = 0;
+  }
+
+  public get length(): number {
+    return this._length;
+  }
+
+  public get capacity(): number {
+    return this._data.length;
+  }
+
+  public ensureCapacity(newCapacity: number): number {
+    if (this.capacity >= newCapacity)
+      return this.capacity;
+
+    newCapacity *= this._growthFactor;
+    const prevData = this._data;
+    this._data = new this._constructor(newCapacity);
+    this._data.set(prevData, 0);
```

Comment:
@dassaf4 some potential easy improvements for the various growable array types like GrowableFloat64Array: replace loops in ensureCapacity and clone with calls to `set` and consider a growth factor closer to 1.5 than 2.

Reply:
Nice! Good idea. Will do this.